src/data/mystery-encounters/utils/encounter-phase-utils.ts

```diff
@@ -962,21 +961,11 @@ export function handleMysteryEncounterBattleStartEffects() {
   ) {
     const effects = encounter.startOfBattleEffects;
     effects.forEach((effect) => {
-      let source;
+      let source: Pokemon;
       if (effect.sourcePokemon) {
         source = effect.sourcePokemon;
       } else if (!isNil(effect.sourceBattlerIndex)) {
```

Comment:
```suggestion
      } else if (!isNil(effect.sourceBattlerIndex) && FIELD_BATTLER_INDEXES.contains(effect.sourceBattlerIndex as FieldBattlerIndex)) {
```
Will also need the imports updated.

Unless it's not possible for `BattlerIndex.ATTACKER` (or any of the other "fake" `BattlerIndex` values) to be in `effect.sourceBattlerIndex`?

Reply:
`sourceBattlerIndex` is now a `FieldBattlerIndex`, so this is already enforced